spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java

```diff
@@ -1,36 +1,25 @@
 package org.springframework.cloud.stream.binder.kinesis.config;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
-import org.springframework.boot.autoconfigure.context.PropertyPlaceholderAutoConfiguration;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
-import org.springframework.cloud.stream.binder.Binder;
 import org.springframework.cloud.stream.binder.kinesis.KinesisMessageChannelBinder;
 import org.springframework.cloud.stream.binder.kinesis.properties.KinesisBinderConfigurationProperties;
 import org.springframework.cloud.stream.binder.kinesis.properties.KinesisExtendedBindingProperties;
-import org.springframework.cloud.stream.binder.kinesis.provisioning.KinesisTopicProvisioner;
-import org.springframework.cloud.stream.config.codec.kryo.KryoCodecAutoConfiguration;
+import org.springframework.cloud.stream.binder.kinesis.provisioning.KinesisStreamProvisioner;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.integration.codec.Codec;
 
 /**
  * 
  * @author Peter Oates
  *
  */
-@Configuration
-@ConditionalOnMissingBean(Binder.class)
-@Import({KryoCodecAutoConfiguration.class, PropertyPlaceholderAutoConfiguration.class})
+//@Configuration
+//@ConditionalOnMissingBean(Binder.class)
+//@Import({KryoCodecAutoConfiguration.class, PropertyPlaceholderAutoConfiguration.class})
 @EnableConfigurationProperties({KinesisBinderConfigurationProperties.class, KinesisExtendedBindingProperties.class})
```

Comment:
You need `@Configuration` to let Spring Container to scan this class properly.
You don't need `@ConditionalOnMissingBean(Binder.class)` because it's not an auto-configuration and there can't be any custom `Binder` in the classpath. If that, the SCSt has some option to deal with multi binder in the CP.

You don't need that `@Import` because it is there per se by Spring Boot auto-config.

So, just remove what must not be here.

And right: that is wrong on Spring Kafka Binder as well if you are going to point me there 😄